Sources/AsyncHTTPClient/AsyncAwait/HTTPClient+execute.swift

```diff
@@ -112,46 +126,76 @@ extension HTTPClient {
                 return response
             }
 
+            resendCount += 1
             currentRequest = newRequest
         }
     }
 
     private func executeCancellable(
         _ request: HTTPClientRequest.Prepared,
         deadline: NIODeadline,
-        logger: Logger
+        logger: Logger,
+        context: ServiceContext,
+        resendCount: Int?
     ) async throws -> HTTPClientResponse {
         let cancelHandler = TransactionCancelHandler()
 
-        return try await withTaskCancellationHandler(
-            operation: { () async throws -> HTTPClientResponse in
-                let eventLoop = self.eventLoopGroup.any()
-                let deadlineTask = eventLoop.scheduleTask(deadline: deadline) {
-                    cancelHandler.cancel(reason: .deadlineExceeded)
-                }
-                defer {
-                    deadlineTask.cancel()
-                }
-                return try await withCheckedThrowingContinuation {
-                    (continuation: CheckedContinuation<HTTPClientResponse, Swift.Error>) -> Void in
-                    let transaction = Transaction(
-                        request: request,
-                        requestOptions: .fromClientConfiguration(self.configuration),
-                        logger: logger,
-                        connectionDeadline: .now() + (self.configuration.timeout.connectionCreationTimeout),
-                        preferredEventLoop: eventLoop,
-                        responseContinuation: continuation
-                    )
-
-                    cancelHandler.registerTransaction(transaction)
-
-                    self.poolManager.executeRequest(transaction)
-                }
-            },
-            onCancel: {
-                cancelHandler.cancel(reason: .taskCanceled)
+        return try await withSpan(request.head.method.rawValue, context: context, ofKind: .client) { span in
```

Comment:
LGTM on using just the method as low cardinality name ([matches otel well](https://opentelemetry.io/docs/specs/semconv/http/http-spans/#name)) 👍